devtools/dump-gossipstore.c

```diff
@@ -0,0 +1,76 @@
+#include <ccan/crc/crc.h>
+#include <ccan/err/err.h>
+#include <common/type_to_string.h>
+#include <common/utils.h>
+#include <fcntl.h>
+#include <gossipd/gen_gossip_store.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[])
+{
+	int fd;
+	u8 version;
+	beint32_t belen, becsum;
+
+	setup_locale();
+
+	if (argc > 2)
+		errx(1, "Need the filename of a gossip store, or stdin");
+
+	if (argv[1]) {
+		fd = open(argv[1], O_RDONLY);
+		if (fd < 0)
+			err(1, "Opening %s", argv[1]);
```

Comment:
Adding the error message might be useful here.

Reply:
err() does that, errx() is the one which doesn't.  Yeah, not my invention :)